.github/workflows/format-typing-check.yml

```diff
@@ -0,0 +1,50 @@
+name: Check format and static typing
+
+on:
+  push:
+    branches:
+      - dev
+      - main
+    paths:
+      - '**.py'
+      - '.github/workflows/format-typing-check.yml'
+  pull_request:
+    branches:
+      - dev
+      - main
+    paths:
+      - '**.py'
+      - '.github/workflows/format-typing-check.yml'
+    types:
+      - opened
+      - reopened
+      - synchronize
+      - ready_for_review
+
+jobs:
+  format-check:
+    name: Check code format
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install ruff
+        run: |
+          pip install ruff mypy types-toml types-Deprecated typing_extensions
```

Comment:
Shouldn't such dependencies be installed via an extra dependencies list defined in the toml file as 'dev', or 'linting' or similar?

Reply:
As we discussed in person, we won't do that because it will make creating development environment more complex.